test/typescript/document.ts

```diff
@@ -15,4 +15,10 @@ void async function main() {
 
   const p: Promise<ITest> = doc.remove();
   await p;
+
+}();
+
+void async function run() {
+  const user = new Test({ name: {}, address: {} });
+  user.validateSync();
```

Comment:
This doesn't test anything. You'll want to do `const error = user.validateSync()`, and then try accessing `error.errors.address.errors`, which should fail without the fix in this PR.

Reply:
Doesn't main.test.js check for errors though? if something in document.ts throws an error, doens't main.test.js store it and do an assertion if there are no errors?